src/utilities/caching/__tests__/getMemoryInternals.ts

```diff
@@ -0,0 +1,173 @@
+        fragmentRegistryLookup: 1000,
+        fragmentRegistryTransform: 2000,
+        getVariableDefinitions: 2000,
+        maybeBroadcastWatch: 5000,
+        parser: 1000,
+        persistedQueryHashes: 2000,
+        print: 2000,
+        queryManagerTransforms: 2000,
+      },
+      sizes: {
+        cache: {
+          addTypenameTransform: [0],
+          fragmentQueryDocuments: 0,
+          fragmentRegistry: {
+            findFragmentSpreads: 0,
+            lookup: 0,
+            transform: 0,
+          },
+          maybeBroadcastWatch: 0,
+          storeReader: {
+            executeSelectionSet: 0,
+            executeSubSelectedArray: 0,
+          },
+        },
+        global: {
+          canonicalStringify: 0,
+          parser: 0,
+          print: 0,
+        },
+        links: [
+          {
+            persistedQueryHashes: 0,
+          },
+          {
+            getVariableDefinitions: 0,
+          },
+        ],
+        queryManager: {
+          Transforms: 0,
+          documentTransforms: [0, 0],
+        },
+      },
+    })
+  );
+});
+
+it("returns information about cache usage (some query triggered)", () => {
+  const client = new ApolloClient({
+    documentTransform: new DocumentTransform((x) => x, {
+      cache: true,
+    }).concat(
+      new DocumentTransform((x) => x, {
+        cache: true,
+      })
+    ),
+    cache: new InMemoryCache({
+      fragments: createFragmentRegistry(),
+    }),
+    link: createPersistedQueryLink({
+      sha256,
+    })
+      .concat(removeTypenameFromVariables())
+      .concat(ApolloLink.empty()),
+  });
+
+  client.query({
+    query: gql`
+      query {
+        hello
+      }
+    `,
+  });
+  expect(client.getMemoryInternals?.()).toEqual(
+    expect.objectContaining({
+      limits: {
+        canonicalStringify: 1000,
+        documentTransform: 2000,
+        executeSelectionSet: 10000,
+        executeSubSelectedArray: 5000,
+        fragmentQueryDocuments: 1000,
+        fragmentRegistryFindFragmentSpreads: 4000,
+        fragmentRegistryLookup: 1000,
+        fragmentRegistryTransform: 2000,
+        getVariableDefinitions: 2000,
+        maybeBroadcastWatch: 5000,
+        parser: 1000,
+        persistedQueryHashes: 2000,
+        print: 2000,
+        queryManagerTransforms: 2000,
+      },
+      sizes: {
+        cache: {
+          addTypenameTransform: [1],
+          fragmentQueryDocuments: 0,
+          fragmentRegistry: {
+            findFragmentSpreads: 1,
+            lookup: 0,
+            transform: 1,
+          },
+          maybeBroadcastWatch: 0,
+          storeReader: {
+            executeSelectionSet: 1,
+            executeSubSelectedArray: 0,
+          },
+        },
+        global: {
+          canonicalStringify: 0,
+          parser: 0,
+          print: 1,
+        },
+        links: [
+          {
+            persistedQueryHashes: 1,
+          },
+          {
+            getVariableDefinitions: 0,
```

Comment:
I'd be curious if you'd be interested in a sort of namespacing or identification for these links. Since these memory profiles will be useful for dev tools, its going to be really difficult to correlate these numbers with the links they belong to. Perhaps something like this?

```suggestion
            'removeTypenameFromVariables.getVariableDefinitions': 0,
```

or 

```suggestion
            removeTypenameFromVariables: {
              getVariableDefinitions: 0,
            }
```

or maybe it makes sense to add a `name` property 🤔 

```suggestion
            name: 'removeTypenameFromVariables',
            getVariableDefinitions: 0,
```

Thoughts? Or does this clutter this up too much?

Reply:
Let's wait for the pattern we come up over in the `cache` PR then we'll adjust here :)